Updates git-python: Visualização de Dados

SAMXPS opened the pull request:
##  Updates git-python: Visualização de Dados
Pull request para as modificações feitas pelo Grupo de Visualização de Dados.

## Colaboradores
- Leonardo Maffei  @maffei2443
- Manuela Matos    @munak98
- Samuel James      @SAMXPS
- Sérgio Freitas       @DasGeist

## Principais Modificações

- Modularização e Separação do [notebook de visualização](../blob/master/python/visualizacao.ipynb) em vários arquivos
- Criação de um [notebook com padrões e técnicas de visualização](https://github.com/SAMXPS/ml-kaggle/blob/master/python/visualizacao/casos.ipynb)
- Documentação em [resumo sobre Matplotlib](https://github.com/SAMXPS/ml-kaggle/blob/master/python/visualizacao/matplotlib.ipynb)
- Documentação em [resumo sobre Seaborn](https://github.com/SAMXPS/ml-kaggle/blob/master/python/visualizacao/seaborn.ipynb)
- [Miscelânea com matplotlib](https://github.com/SAMXPS/ml-kaggle/blob/master/python/visualizacao/miscelanea.ipynb)
- Adicionado exemplos de [gráficos tridimensionais](https://github.com/SAMXPS/ml-kaggle/blob/master/python/visualizacao/tridimensionais.ipynb) com matplotlib

## TODO:

Apesar das alterações já estarem em bom estado e terem sido brevemente revisadas, ainda nos falta finalizar:
- [x] Estruturação de links.
- [x] Sugerir uma ordem de estudo no [notebook principal](../blob/master/python/visualizacao.ipynb).
- [ ] Corrigir sugestão de estudos segundo [sugestão](https://github.com/UnB-CIC/ml-kaggle/pull/2#commitcomment-38468231) de @maffei2443

Sugestões de @gnramos :
- [x] Ajustar o link para o novo diretório "visualizacao" no README
- [x] Reestruturar parte do texto em células, isolando os assuntos. Visualmente inútil, mas facilita futuras alterações.
- [x] Ajustar link para programacao.ipynb (não funciona).
- [x] Incorporar o https://github.com/matplotlib/ipympl (ou incluir no "miscelanea")
- [ ] Rever o uso de blockquotes ">" para indicar links úteis, a ideia desta formatação é mais apresentar um "grande" conteúdo de texto citado, não destaca um link externo.
- [ ] (Parcialmente Solucionado) Seções de "Referência" não fazem muito sentido neste contexto. As fontes certamente precisam ser citadas, mas podem ser somente indicadas com os devidos hyperlinks nos trechos específicos. Ex: em miscelanea.ipynb, a referência poderia sera dada no título da seção: "### Desenhar uma seta"

maffei2443 committed:
update visualizacao.ipynb

SAMXPS committed:
Introdução à visualização de dados

SAMXPS committed:
Exemplo de figure com vários axes e representação de gráficos

SAMXPS committed:
Títulos e Legendas no matplotlib

SAMXPS committed:
Removendo gráfico "Olá Mundo"

Acredito que não estava inserido no contexto correto, uma vez que já tinham sido introduzidos outros exemplos mais concretos

SAMXPS committed:
Gráficos em 3D com o matplotlib

maffei2443 committed:
add notebook feito pela manu

maffei2443 committed:
Merge branch 'master' of https://github.com/SAMXPS/ml-kaggle

maffei2443 committed:
solving conflicts

SAMXPS committed:
Modularizando parte de visualização

SAMXPS committed:
Incluindo notebook da manu no seaborn

maffei2443 committed:
finish [seaborn] scatterplot

maffei2443 committed:
melhora seaborn, add boxplot

SAMXPS committed:
Introdução aos gráficos de dispersão em 3 dimensões

aMelliflousOne committed:
Heatmaps e correlação

aMelliflousOne committed:
Merge manual

SAMXPS committed:
Adicionando introdução à visualização de dados em Aprendizado de Máquina

maffei2443 committed:
seaborn little better

maffei2443 committed:
fix conflict

aMelliflousOne committed:
Guia de técnicas de visualização

aMelliflousOne committed:
Merge branch 'master' of https://github.com/SAMXPS/ml-kaggle

aMelliflousOne committed:
Correções de português (perdão).

maffei2443 committed:
.

SAMXPS committed:
Merge remote-tracking branch 'upstream/master'

aMelliflousOne committed:
Reestruturação da raíz de visualização

maffei2443 committed:
last.

maffei2443 committed:
Merge branch 'master' of https://github.com/SAMXPS/ml-kaggle

SAMXPS committed:
Merge remote-tracking branch 'upstream/master'

aMelliflousOne committed:
Adicionei um link para miscelanea.ipynb

SAMXPS committed:
Merge remote-tracking branch 'upstream/master'

SAMXPS committed:
Adicionando "links amigáveis" ao notebook sobre matplotlib

SAMXPS committed:
Sugestão de estudo sobre visualização de dados

SAMXPS committed:
Alterando link de sugestão final com sintaxe markdown

SAMXPS committed:
Removendo headers e tornando-os notes

SAMXPS committed:
Update broken links

gnramos commented:
0. No aguardo da versão "final".
1. Ajustar o link para o vo diretório "visualizacao" no README
2. Reestruturar parte do texto em células, isolando os assuntos. Visualmente inútil, mas facilita futuras alterações.
3. Ajustar link para programacao.ipynb (não funciona).
4. Incorporar o https://github.com/matplotlib/ipympl (ou incluir no "miscelanea")
5. Rever o uso de blockquotes ">" para indicar links úteis, a ideia desta formatação é mais apresentar um "grande" conteúdo de texto citado, não destaca um link externo.
6. Seções de "Referência" não fazem muito sentido neste contexto. As fontes *certamente* precisam ser citadas, mas podem ser somente indicadas com os devidos hyperlinks nos trechos específicos. Ex: em miscelanea.ipynb, a referência poderia sera dada no título da seção: "### Desenhar uma [seta](https://matplotlib.org/3.2.1/gallery/text_labels_and_annotations/arrow_simple_demo.html#sphx-glr-gallery-text-labels-and-annotations-arrow-simple-demo-py)"

maffei2443 commented:
Adicionar como contribuidor @munak98, a manu colaborou antes de sair da disciplina

SAMXPS commented:
> 2. Ajustar o link para o vo diretório "visualizacao" no README

@gnramos em qual README exatamente?
Diz para criarmos um novo README dentro do diretório visualizacao?

SAMXPS commented:
> Adicionar como contribuidor @munak98, a manu colaborou antes de sair da disciplina

Feito.

SAMXPS committed:
Ajustar link para programacao.ipynb (não funcionava)

SAMXPS reviewed (commented):
Solucionando problema #3 do [pull request](https://github.com/UnB-CIC/ml-kaggle/pull/2#issuecomment-612911759)

SAMXPS committed:
Solucionando problema #2 do [pull request](https://github.com/UnB-CIC/ml-kaggle/pull/2#issuecomment-612911759)

Reestruturar parte do texto em células, isolando os assuntos. Visualmente inútil, mas facilita futuras alterações.

SAMXPS committed:
Removendo parte das referências "que não fazem muito sentido" no contexto dos notebooks

Modifiacações baseadas na sugestão de @gnramos https://github.com/UnB-CIC/ml-kaggle/pull/2#issuecomment-612911759

SAMXPS commented:
Vou marcar este pull request como "Ready for review"

Acredito que ainda poderemos fazer alterações nele.

SAMXPS committed:
Teste com ipympl

SAMXPS committed:
Reconfigurando saídas para funcionarem no GitHub

gnramos commented:
> @gnramos em qual README exatamente?
> Diz para criarmos um novo README dentro do diretório visualizacao?
É o noção de [programação em python](/programacao.ipynb) (em python/visualizacao.ipynb). Esse link é absoluto e quebrado se inicio o jupyter no diretório "python". provavelmente retirando a "/" inicial resolve. 


> @gnramos @lpfgarcia
> O que acham sobre esta sugestão dentro do texto?
> 
> > Encorajamos os estudantes a submenterem suas próprias contribuições por meio de forks e pull-requests

"submeterem". Acho ótimo, melhor ainda se puderem exemplificar algumas contribuições possíveis/desejáveis. Outra coisa legal é gerar issues para melhorias (e rotulando-as com o tag apropriado).

maffei2443 commented:
@gnramos qual a necessidade de haver um link apontando para o diretório **visualiacao**? A ideia do **visualização.ipynb** não é justamente servir como o ponto de partida para encontrar os notebooks dessa pasta?

maffei2443 commented:
@gnramos quanto a hava uma seção de referências, particularmente me agrada a idea pelo fato de reunir em um só local, senão todas, diversas fontes utilizadas ao longo da confecção do caderno. Por exemplo, por diversas vezes após seguir algum exemplo do **matplotlib** me vi acessando outros links, como a documentação de classes. Isso não é essencial para muitas das funcionalidades básicas conforme os tutoriais da documentação mostra, mas eu usei então teoricamente deveria colocar essa informação em algum lugar.

Mas, se for de seu agrado essa remoção, sem problemas.

maffei2443 committed:
padroniza anotação dos links; insere links que estavam apenas nas referência para onde eles serviram de fonte

SAMXPS committed:
Corrigindo typo
